crates/bevy_ecs/src/schedule_v3/condition.rs

```diff
@@ -0,0 +1,100 @@
+pub use helper::*;
+
+use crate::system::BoxedSystem;
+
+pub type BoxedCondition = BoxedSystem<(), bool>;
+
+/// A system that determines if one or more scheduled systems should run.
+///
+/// Implemented for functions and closures that convert into [`System<In=(), Out=bool>`](crate::system::System)
+/// with [read-only](crate::system::ReadOnlySystemParamFetch) parameters.
+pub trait Condition<Params>: sealed::Condition<Params> {}
+
+impl<Params, F> Condition<Params> for F where F: sealed::Condition<Params> {}
+
+mod sealed {
+    use crate::system::{
+        IntoSystem, IsFunctionSystem, ReadOnlySystemParamFetch, SystemParam, SystemParamFunction,
+    };
+
+    pub trait Condition<Params>: IntoSystem<(), bool, Params> {}
+
+    impl<Params, Marker, F> Condition<(IsFunctionSystem, Params, Marker)> for F
+    where
+        F: SystemParamFunction<(), bool, Params, Marker> + Send + Sync + 'static,
+        Params: SystemParam + 'static,
+        Params::Fetch: ReadOnlySystemParamFetch,
+        Marker: 'static,
+    {
+    }
+}
+
+pub mod helper {
+    use crate::schedule_v3::{State, Statelike};
+    use crate::system::{Res, Resource};
+
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that returns `true`
+    /// if the resource exists.
+    pub fn resource_exists<T>() -> impl FnMut(Option<Res<T>>) -> bool
```

Comment:
Can we also have `run_unless_resource_exists` as in `iyes_loopless`?

Reply:
It's not clear to me that this will be broadly useful?

Reply:
How about a method on `Condition` that inverts the condition like `.pipe(|In(b)| !b)` would?

Just tried this but `.run_if((|| false).pipe(|In(b)| !b))` does not work as a PipeSystem can't be used as a condition :(

Reply:
Yep, I'd love a general inversion method.

Reply:
If someone run something on `resource_exists` it may also need to run something on `unless_resource_exists`. I don't see any harm having more general purpose methods, it just a few lines, but improves ergonomic a lot.

Reply:
Made an issue for that; we can punt to a new PR.